bin/propolis-server/src/lib/server.rs

```diff
@@ -551,21 +551,27 @@ async fn instance_ensure_common(
     // Since `block_on` will panic if called from an async context, as we are in
     // now, the whole process is wrapped up in `spawn_blocking`.  It is
     // admittedly a big kludge until this can be better refactored.
-    let vm = {
+    let vm = 'create: {
         let properties = properties.clone();
         let server_context = server_context.clone();
         let log = server_context.log.clone();
-        let hdl = tokio::runtime::Handle::current();
-        let ctrl_hdl = hdl.clone();
-        let vm_hdl = hdl.spawn_blocking(move || {
+
+        let vmm_rt = match VmController::spawn_runtime(&properties) {
+            Ok(rt) => rt,
+            Err(e) => break 'create Err(e),
+        };
+
+        // Block for VM controller setup under the current (API) runtime
+        let cur_rt_hdl = tokio::runtime::Handle::current();
+        let vm_hdl = cur_rt_hdl.spawn_blocking(move || {
             VmController::new(
```

Comment:
Take it or leave it: we could avoid the labeled `break` if we rolled the runtime construction into `VmController::new` and just had that be one fallible operation --- AFAICT there oughtn't be anywhere where we need to separate constructing the VMM runtime and constructing the actual `VmController` struct?

Reply:
Done.